packages/reaction-core/common/methods/cart.js

```diff
@@ -4,8 +4,9 @@
 //
 if (Meteor.isClient) {
   Meteor.methods({
-    "cart/submitPayment": function (paymentMethod) {
+    "cart/submitPayment": function (paymentMethod, sessionId) {
```

Comment:
we need `sessionId` in `carts/cartCopyToOrder`. To get it from client the only way I found is to pass it via `cart/submitPayment` hook. We even not use `SessionId` directly inside `cart/submitPayment`. I'm not sure I understand meaning of your proposal, but if we remove SessionId from `cart/submitPayment` arguments we will not get SessionId in `cart/mergeCart`